src/main/java/games/strategy/triplea/settings/SelectionComponentFactory.java

```diff
@@ -340,11 +344,26 @@ public void reset() {
     return selectFile(clientSetting, SwingComponents.FolderSelectionMode.DIRECTORIES);
   }
 
-  static Supplier<SelectionComponent> selectionBox(
+  static <T> Supplier<SelectionComponent> selectionBox(
       final ClientSetting clientSetting,
-      final List<String> availableOptions) {
+      final List<T> availableOptions,
+      final Function<T, ? extends Component> renderFunction) {
```

Comment:
The ability to specify an arbitrary renderer component may be a YAGNI.  It might be simpler at this time to just have the user specify a `Function<T, String>` that produces the text that should be displayed and manage the renderer component internally to this method.